src/screens/BlaireHome/BlaireHome.tsx

```diff
@@ -1,20 +1,130 @@
+import type {ImageSourcePropType, ListRenderItem} from 'react-native';
 import * as React from 'react';
-import {useCallback} from 'react';
-import {Linking, Text, View} from 'react-native';
+import {useCallback, useEffect, useState} from 'react';
+import {Alert, FlatList, Linking, Text, View} from 'react-native';
 import Button from 'components/Button';
-
+import SingleCard from './_private/SingleCard';
 import styles from './styles';
 
+const NUM_COLS = 4;
+const NUM_PAIRS = 6;
+
+const cardImages: Card[] = [
+  {matched: false, src: require('./assets/blue.png')},
+  {matched: false, src: require('./assets/red.png')},
+  {matched: false, src: require('./assets/orange.png')},
+  {matched: false, src: require('./assets/green.png')},
+  {matched: false, src: require('./assets/yellow.png')},
+  {matched: false, src: require('./assets/purple.png')},
+];
+
+type Card = {
+  id?: number;
+  src: ImageSourcePropType;
+  matched: boolean;
+};
+
 type Props = {};
 
 function BlaireHome(props: Props) {
-  const onPress = useCallback(() => {
+  const [cards, setCards] = useState<Card[]>([]);
+  const [turns, setTurns] = useState<number>(0);
+  const [pairs, setPairs] = useState<number>(NUM_PAIRS);
```

Comment:
it seems like `NUM_PAIRS` is pretty disconnected from the actual number of pairs generated, perhaps you should just initialize this value to `cardImages.length`?